client/src/components/App/Content/InvestigationForm/TabManagement/useTabManagement.ts

```diff
@@ -1,49 +1,71 @@
+
+    const lastTabDisplayedId = areThereContacts ? LAST_TAB_ID : LAST_TAB_ID - 1;
 
     const [currentTab, setCurrentTab] = useState<number>(0);
     const nextTab: RefObject<number> = useRef<number>(0);
     const epidemiologyNumber = useSelector<StoreStateType, number>(store => store.investigation.epidemiologyNumber);
-    const currentInvestigationValidations = useSelector<StoreStateType, (boolean | null)[]>(store => store.formsValidations[epidemiologyNumber]);
-    
+    const tabsValidations = useSelector<StoreStateType, (boolean | null)[]>(store => store.formsValidations[epidemiologyNumber]);
+
     useEffect(() => {
-        if (currentInvestigationValidations !== undefined) {
-            const areTabsNotValid: boolean = validTabsBeforeExport.some(tabId => !currentInvestigationValidations[tabId]);
-            if (exportToMabarTabs.includes(nextTab.current as number)) {
-                if (areTabsNotValid) {
-                    alertError('חלק מן השדות אינם תקינים, נא מלא אותם מחדש ונסה שוב.', {
-                        text: 'שים לב שלא עברת בחלק מהטאבים'
-                    });
-                    setNextTab(currentTab);
+        if (tabsValidations !== undefined) {
+            const areTabsNotValid: boolean = validTabsBeforeExport.some(tabId => !tabsValidations[tabId]);
+            if (exportToMabarTabs.includes(nextTab.current as number) && areTabsNotValid) {
+                alertError('חלק מן השדות אינם תקינים, נא מלא אותם מחדש ונסה שוב.', {
+                    text: 'שים לב שלא עברת בחלק מהטאבים'
+                });
+                setNextTab(currentTab);
+            } else {
+                if(nextTab.current === lastTabDisplayedId + 1) {
+                    if(isInvestigationValid()) {
+                        confirmFinishInvestigation(epidemiologyNumber);
+                    } else {
+                        const displayedForms = [...tabsValidations];
+                        !areThereContacts && displayedForms.splice(orderedTabsNames.findIndex(tabName => tabName === TabNames.CONTACT_QUESTIONING), 1);
+                        const didAnyTabsWasntChecked = displayedForms.some(form => form === null);
+                        alertError('חלק מן השדות אינם תקניים, נא מלא אותם מחדש ונסה שוב.', {
+                            text: didAnyTabsWasntChecked ? 'שים לב שלא עברת בחלק מהטאבים' : '',
+                        })
+                    }
                 } else {
                     setCurrentTab(nextTab.current as number);
                 }
-            } else {
-                setCurrentTab(nextTab.current as number);
             }
         }
-    }, [currentInvestigationValidations])
+    }, [tabsValidations])
 
     const setNextTab = (nextTabId: number) => {
         //@ts-ignore
         nextTab.current = nextTabId;
     }
 
+    const isInvestigationValid = () => {
+        return !(tabsValidations.slice(0, lastTabDisplayedId + 1).some((tabValidation) => !tabValidation));
```

Comment:
why you need to slice the tabsValidations array?

Reply:
we just moved it there as it was

Reply:
I guess it because we don't know the length of the array. it depends on areThereContacts variable